projects/igniteui-angular/src/lib/directives/focus-trap/focus-trap.directive.ts

```diff
@@ -0,0 +1,108 @@
+import { AfterViewInit, Directive, ElementRef, Input, NgModule, OnDestroy } from '@angular/core';
+import { fromEvent, Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+import { PlatformUtil } from '../../core/utils';
+
+@Directive({
+    selector: '[igxFocusTrap]'
+})
+export class IgxFocusTrapDirective implements AfterViewInit, OnDestroy {
+    /** @hidden */
+    public get element(): HTMLElement | null {
+        return this.elementRef.nativeElement;
+    }
+
+    private destroy$ = new Subject();
+    private _focusTrap: boolean;
+
+    /** @hidden */
+    constructor(
+        private elementRef: ElementRef,
+        protected platformUtil: PlatformUtil) {
+    }
+
+    /**
+     * Sets the type of the button.
+     *
+     * @example
+     * ```html
+     * <button igxButton="icon"></button>
+     * ```
+     */
+    @Input('igxFocusTrap')
+    public set focusTrap(focusTrap: boolean) {
+        this._focusTrap = focusTrap || true;
+    }
+
+    /** @hidden */
+    public ngAfterViewInit(): void {
+        if (this._focusTrap) {
```

Comment:
this way if `focusTrap` is changed at run time component will not know about this. Either we should put the check in the subscription (this way even if `focusTrap` is false we will subscribe), or subscription should be added/removed in the `focusTrap` setter. IMO we may go with first option - check `focustTrap` in the subscription.

Reply:
you can add this scenario to tests too